src/html/index.html

```diff
@@ -132,22 +132,46 @@ <h3>Name: {{ stream.video_and_stream.name }}</h3>
                     return undefined
                 },
                 configureStream: async function(stream) {
-                    const content = {
-                        "name": stream.name,
-                        "source": stream.source,
-                        "stream_information": {
-                            "endpoints": stream.endpoints ? stream.endpoints.split(',') : ["udp://0.0.0.0:5600"],
-                            "configuration": {
-                                "encode": stream.configuration.encode,
-                                "height": Number(stream.configuration.size.height),
-                                "width": Number(stream.configuration.size.width),
-                                "frame_interval": stream.configuration.interval,
-                            },
-                            "extended_configuration": {
-                                "thermal": Boolean(stream.configuration.thermal),
-                            },
+                    const content = (() => {
+                        if (stream.source == "Redirect") {
+                            return {
```

Comment:
No problem, I was playing with ways of doing that change keeping the variable as _const_, and just kept this to see your opinion. I didn't like it this way either. There are three other options that I am aware of:
1. Using let instead of const and changing the specific fields after the declaration:
```javascript
                    let content = {
                        "name": stream.name,
                        "source": stream.source,
                        "stream_information": {
                            "endpoints": stream.endpoints ? stream.endpoints.split(',') : ["udp://0.0.0.0:5600"],
                            "configuration": {
                                "encode": stream.configuration.encode,
                                "height": Number(stream.configuration.size.height),
                                "width": Number(stream.configuration.size.width),
                                "frame_interval": stream.configuration.interval,
                            },
                            "extended_configuration": {
                                "thermal": Boolean(stream.configuration.thermal),
                            },
                    }
                    if (stream.source == "Redirect") {
                        content.stream_information.configuration.encode = "H264";
                        content.stream_information.configuration.height = 0;
                        content.stream_information.configuration.width = 0;
                        content.stream_information.configuration.frame_interval.numerator = 0;
                        content.stream_information.configuration.frame_interval.denominator = 0;
                    }

```
2. Use variables to assign each field:
```javascript
                    let encode = stream.configuration.encode
                    let height = stream.configuration.size.height
                    let width = stream.configuration.size.width
                    let frame_interval = stream.configuration.interval

                    if (stream.source == "Redirect") {
                        encode = "H264"
                        height = 0
                        width = 0
                        frame_interval.numerator = 0
                        frame_interval.denominator = 0
                    }

                    const content = {
                        "name": stream.name,
                        "source": stream.source,
                        "stream_information": {
                            "endpoints": stream.endpoints ? stream.endpoints.split(',') : ["udp://0.0.0.0:5600"],
                            "configuration": {
                                "encode": encode,
                                "height": height,
                                "width": width,
                                "frame_interval": frame_interval,
                            },
                            "extended_configuration": {
                                "thermal": Boolean(stream.configuration.thermal),
                            },
                        },
                    }
```
3. Use a ternary operation on each field:
```javascript
                    const redirect = stream.source == "Redirect"
                    const content = {
                        "name": stream.name,
                        "source": stream.source,
                        "stream_information": {
                            "endpoints": stream.endpoints ? stream.endpoints.split(',') : ["udp://0.0.0.0:5600"],
                            "configuration": {
                                "encode": redirect ? "H264" : stream.configuration.encode,
                                "height": redirect ? 0 : Number(stream.configuration.size.height),
                                "width": redirect ? 0 : Number(stream.configuration.size.width),
                                "frame_interval": redirect ? {
                                    numerator: 0,
                                    denominator: 0,
                                } : stream.configuration.interval,
                            },
                            "extended_configuration": {
                                "thermal": Boolean(stream.configuration.thermal),
                            },
                        },
                    }
```

Which one do you prefer?

Reply:
Why do you want to force H264 encode for redirect ? and why this information is important ? it should be none or unknown

Reply:
later I'll rewrite this part in some less annoying way.

Reply:
Can we do it without the else ?